colour/algebra/interpolation.py

```diff
@@ -31,23 +37,598 @@
+    def x(self):
+        """
+        Getter and setter property for the independent :math:`x` variable.
+
+        Parameters
+        ----------
+        value : array_like
+            Value to set the independent :math:`x` variable with.
+
+        Returns
+        -------
+        array_like
+            Independent :math:`x` variable.
+        """
+
+        return self._x
+
+    @x.setter
+    def x(self, value):
+        """
+        Setter for the **self.x** property.
+        """
+
+        if value is not None:
+            value = np.atleast_1d(value).astype(self._dtype)
+
+            assert value.ndim == 1, (
+                '"x" independent variable must have exactly one dimension!')
+
+            value_interval = interval(value)
+
+            if value_interval.size != 1:
+                warning(('"x" independent variable is not uniform, '
+                         'unpredictable results may occur!'))
+
+            self._x = value
+
+            if self._window is not None:
+                self._x_p = np.pad(
+                    self._x, (self._window, self._window),
+                    'linear_ramp',
+                    end_values=(
+                        np.min(self._x) - self._window * value_interval[0],
+                        np.max(self._x) + self._window * value_interval[0]))
+
+    @property
+    def y(self):
+        """
+        Getter and setter property for the dependent and already known
+        :math:`y` variable.
+
+        Parameters
+        ----------
+        value : array_like
+            Value to set the dependent and already known :math:`y` variable
+            with.
+
+        Returns
+        -------
+        array_like
+            Dependent and already known :math:`y` variable.
+        """
+
+        return self._y
+
+    @y.setter
+    def y(self, value):
+        """
+        Setter for the **self.y** property.
+        """
+
+        if value is not None:
+            value = np.atleast_1d(value).astype(self._dtype)
+
+            assert value.ndim == 1, (
+                '"y" dependent variable must have exactly one dimension!')
+
+            self._y = value
+
+            if self._window is not None:
+                self._y_p = np.pad(self._y, **self._padding_args)
+
+    @property
+    def window(self):
+        """
+        Getter and setter property for the half window.
+
+        Parameters
+        ----------
+        value : int
+            Value to set the half window with.
+
+        Returns
+        -------
+        int
+            Half window.
+        """
+
+        return self._window
+
+    @window.setter
+    def window(self, value):
+        """
+        Setter for the **self.window** property.
+        """
+
+        if value is not None:
+            assert is_integer(value), '"window" must be an integer!'
+
+            assert value >= 1, '"window" must be equal or superior to 1!'
+
+            self._window = value
+
+            if self._x is not None:
```

Comment:
If these self-assignments happen for their side effects, should they be wrapped in a function for documentation purposes?

Reply:
It is not exactly a self-assignement and this is part of the protected API people should not touch. We could maybe implement a `KernelInterpolator._pad` protected method to make more obvious what is happening?

Reply:
Okay. Do you think an in-line comment then for future internal changes might be helpful anyway?

Reply:
Yes, I will add one.